### PR TITLE
Cypher should be able to run queries when query cache size is 0

### PR DESCRIPTION
**Pull request by davidegrohmann**

### CODE REVIEW

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/ExecutionEngineIT.scala [hunk 1]
```diff
@@ -73,16 +73,29 @@ class ExecutionEngineIT extends CypherFunSuite {
...
-        .newImpermanentDatabaseBuilder()
-        .setConfig(GraphDatabaseSettings.cypher_planner, "COST")
-        .setConfig(GraphDatabaseSettings.cypher_parser_version, "2.0").newGraphDatabase()
+    val db = new TestGraphDatabaseFactory()
+      .newImpermanentDatabaseBuilder()
+      .setConfig(GraphDatabaseSettings.cypher_planner, "COST")
+      .setConfig(GraphDatabaseSettings.cypher_parser_version, "2.0").newGraphDatabase()
 
+    // when-then
+    intercept[Exception] {
```
Comment (Mats-SX) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/ExecutionEngineIT.scala:82]: I like `an [Exception] should be thrownBy { ... }`

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/ExecutionEngine.scala [hunk 2]
```diff
@@ -140,32 +140,24 @@ class ExecutionEngine(graph: GraphDatabaseService, logger: StringLogger = String
...
       val kernelStatement = txBridge.instance()
 
-      val (plan: ExecutionPlan, extractedParameters) = try {
+      val ((plan: ExecutionPlan, extractedParameters), touched) = try {
         // fetch plan cache
-        val cache: LRUCache[String, (ExecutionPlan, Map[String, Any])] = getOrCreateFromSchemaState(kernelStatement, {
+        val cache = getOrCreateFromSchemaState(kernelStatement, {
           cacheMonitor.cacheFlushDetected(kernelStatement)
-          new LRUCache[String, (ExecutionPlan, Map[String, Any])](getPlanCacheSize)
+          val lruCahche = new LRUCache[String, (ExecutionPlan, Map[String, Any])](getPlanCacheSize)
```
Comment (Mats-SX) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/ExecutionEngine.scala:151]: Misspelled variable

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/helpers/LRUCache.scala [hunk 3]
```diff
@@ -21,10 +21,10 @@ package org.neo4j.cypher.internal.compiler.v2_2.helpers
 
 import com.googlecode.concurrentlinkedhashmap.ConcurrentLinkedHashMap
 
-class LRUCache[K, V](cacheSize: Int) extends ((K, => V) => V) {
+class LRUCache[K, V](val size: Int) extends ((K, => V) => V) {
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/helpers/LRUCache.scala:24]: Isn't the declaration of `val` here redundant?

Comment (systay) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/helpers/LRUCache.scala:24]: Nope. It's used [here](https://github.com/neo4j/neo4j/pull/6569/files#diff-d13b42573e66ee6724d02b58fbf36682R31)

**Pull request merged**